org.eclipse.draw2d/src/org/eclipse/draw2d/geometry/Rectangle.java

```diff
@@ -1050,6 +1060,70 @@ public Rectangle shrink(int h, int v) {
 		return this;
 	}
 
+	/**
+	 * Shrinks the height of the rectangle by the given amount, keeping
+	 * {@link #top()}.
+	 * 
+	 * @param deltaY The vertical reduction amount
+	 * @return <code>this</code> for convenience
+	 * @since 3.13
+	 */
+	public Rectangle shrinkBottom(int deltaY) {
+		height = deltaY - y;
```

Comment:
from the description I would have expected that here `height -= deltaY;' would be the right code. If I understand your code correctly it takes deltaY as an absolute Y coordinate. But this would be different to the other shrink methods.

Reply:
Hm, I think you're right. I suppose I was a little bit too eager to in having every method start with "shrink". `shrinkBottom` and `shrinkRight` allow you set the y-coordinate of the bottom and the x-coordinate side.

Originally, I had the methods named `setBottom` and `setRight`, which I've now went back to. I've also adjusted the checks, to make sure that neither width/height can never become negative.

Reply:
Reading the code I thought that. With the updates the patch is for me ready to merge